### PR TITLE
DEPR: pd.value_counts

### LINKED ISSUE #47862: BUG or DOCS:  Is pd.value_counts() public?  If yes, it needs docs.  If not, it should not get exported.
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
pd.value_counts([1,2,3,2,1])
```

### Issue Description

`pd.value_counts()` works, but it does not appear in the documentation.  

Two possibilities:
1. It should not be exported and is not intended to be available.
2. It should be exported, and needs documentation by modifying `pandas/doc/source/reference/general_functions.rst`

Uncovered when cleaning up `pandas-stubs`:  https://github.com/pandas-dev/pandas-stubs/pull/170#issuecomment-1195381657

### Expected Behavior

Unclear!

### Installed Versions

pandas 1.4.3 and associated docs

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #47862 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: doc/source/whatsnew/v0.19.2.rst [hunk 1]
```diff
@@ -7,6 +7,7 @@ Version 0.19.2 (December 24, 2016)
 
 .. ipython:: python
    :suppress:
+   :okwarning:
```
Comment (mroeschke) [inline doc/source/whatsnew/v0.19.2.rst:10]: So a star import will raise the `FutureWarning` from `pd.value_counts`? That's kinda unfortunate as it seems noisy

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.19.2.rst:10]: i dont see an alternative that isnt behavior-changing

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.19.2.rst:10]: i guess could warn inside value_counts if it is called

Comment (mroeschke) [reply doc/source/whatsnew/v0.19.2.rst:10]: Yeah I think it's more directly useful to warn users who are using `pd.value_counts` directly`

Comment (jbrockmendel) [reply doc/source/whatsnew/v0.19.2.rst]: updated + greenish

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**